3-0-spring-framework/3-1-1-dispatcher-servlet-initializer/src/test/java/com/bobocode/WelcomeWebAppTest.java

```diff
@@ -0,0 +1,173 @@
+    void rootConfigComponentScanPackages() {
+        ComponentScan componentScan = RootConfig.class.getAnnotation(ComponentScan.class);
+        String[] packages = componentScan.basePackages();
+        if (packages.length == 0) {
+            packages = componentScan.value();
+        }
+
+        assertThat(packages).contains("com.bobocode");
+    }
+
+    @Test
+    @Order(4)
+    @DisplayName("RootConfig @ComponentScan contains web scan filters")
+    void rootConfigComponentScanFilters() {
+        ComponentScan componentScan = RootConfig.class.getAnnotation(ComponentScan.class);
+        Filter[] filters = componentScan.excludeFilters();
+        List<Class> filteredClasses = getFilteredClasses(filters);
+
+        assertThat(filters.length).isEqualTo(2);
+        assertThat(filters[0].type()).isEqualTo(FilterType.ANNOTATION);
+        assertThat(filters[1].type()).isEqualTo(FilterType.ANNOTATION);
+        assertThat(filteredClasses.toArray()).containsExactlyInAnyOrder(EnableWebMvc.class, Controller.class);
+    }
+
+    @Test
+    @Order(5)
+    @DisplayName("WebConfig is marked as @Configuration")
+    void webConfigIsMarkedAsConfiguration() {
+        Configuration configuration = WebConfig.class.getAnnotation(Configuration.class);
+
+        assertNotNull(configuration);
+    }
+
+    @Test
+    @Order(6)
+    @DisplayName("WebConfig enables @ComponentScan")
+    void webConfigEnablesComponentScan() {
+        ComponentScan componentScan = WebConfig.class.getAnnotation(ComponentScan.class);
+
+        assertNotNull(componentScan);
+    }
+
+    @Test
+    @Order(7)
+    @DisplayName("WebConfig @ComponentScan contains web packages")
+    void webConfigComponentScanPackages() {
+        ComponentScan componentScan = WebConfig.class.getAnnotation(ComponentScan.class);
+
+        assertThat(componentScan.basePackages()).contains("com.bobocode.web");
+    }
+
+    @Test
+    @Order(8)
+    @DisplayName("WebConfig is marked with @EnableWebMvc")
+    void webConfigEnablesWebMvc() {
+        EnableWebMvc enableWebMvc = WebConfig.class.getAnnotation(EnableWebMvc.class);
+
+        assertNotNull(enableWebMvc);
+    }
+
+    @Test
+    @Order(9)
+    @DisplayName("Initializer method is overridden with RootConfig class")
+    void initializerRootConfigClasses() {
+        WebAppInitializerWrapper webAppInitializerWrapper = new WebAppInitializerWrapper();
+
+        assertThat(webAppInitializerWrapper.getRootConfigClasses()).contains(RootConfig.class);
+    }
+
+    @Test
+    @Order(10)
+    @DisplayName("Initializer method is overridden with WebConfig class")
+    void initializerWebConfigClasses() {
+        WebAppInitializerWrapper webAppInitializerWrapper = new WebAppInitializerWrapper();
+
+        assertThat(webAppInitializerWrapper.getServletConfigClasses()).contains(WebConfig.class);
+    }
+
+    @Test
+    @Order(11)
+    @DisplayName("Proper Dispatcher Servlet mapping")
```

Comment:
@mathodcoast `Dispatcher Servlet mapping is configured` or `Dispatcher Servlet mapping is "/"`